### PR TITLE
[Audit] Add bucketed scan info in query plan of data source v1 [databricks]

### LINKED ISSUE #3952: [Audit] [FEA][SPARK-32986][SQL] Add bucketed scan info in query plan of data source v1
**Is your feature request related to a problem? Please describe.**
The PR adds info into `FileSourceScanExec` physical query plan output whether the table is read as bucketed table. Opening this issue to get feedback if we need to update GpuFileSourceScanExec.scala to include this change.
PR - https://github.com/apache/spark/commit/79515e4b6c

### PR DESCRIPTION
**Pull request by HaoYang670**
Signed-off-by: remzi 13716567376yh@gmail.com
close #3952 

1. update GpuFileSourceScanExec.scala to include the change of spark https://github.com/apache/spark/commit/79515e4b6c
2. add an integration test
3. update shim layers from spark31X

### CODE REVIEW

**Comment by HaoYang670** [comment]
build

File: integration_tests/src/main/python/explain_test.py [hunk 1]
```diff
@@ -91,3 +91,53 @@ def do_explain(spark):
 
     with_cpu_session(do_explain)
 
+@allow_non_gpu(any=True)
+def test_explain_bucketed_scan(spark_tmp_table_factory):
+    """
+    https://github.com/NVIDIA/spark-rapids/issues/3952
```
Comment (revans2) [inline integration_tests/src/main/python/explain_test.py:97]: A text explanation would be good to be able to get a high level idea of what is happening.

**Review by revans2 — COMMENTED** [review]
Just some nits. But, we need to understand and fix the failing test. We also should run these against databricks so we are sure it works there too.

File: integration_tests/src/main/python/explain_test.py [hunk 2]
```diff
@@ -91,3 +91,53 @@ def do_explain(spark):
...
+
+    def disable_by_query_planner(spark):
+        tbl = spark_tmp_table_factory.get()
+        spark.createDataFrame([(1, 2), (2, 3)], ("i", "j")).write.bucketBy(8, "i").saveAsTable(tbl)
+        df = spark.table(tbl)
+
+        assert "Bucketed: false (disabled by query planner)" in df._sc._jvm.PythonSQLUtils.explainString(df._jdf.queryExecution(), "simple")
+
+
+    with_gpu_session(bucket_column_not_read)
```
[the remark this answers is not in the document]
Comment (HaoYang670) [reply integration_tests/src/main/python/explain_test.py:139]: Thank you for your feedback! I will split it.

File: integration_tests/src/main/python/explain_test.py (hunk 1 above)
Comment (HaoYang670) [reply integration_tests/src/main/python/explain_test.py:97]: Thank you for the feedback! I will add comments.

**Comment by HaoYang670** [comment]
build

**Comment by HaoYang670** [comment]
Solve merge conflicts because shim layers are refactored.

**Comment by HaoYang670** [comment]
build

**Review by revans2 — COMMENTED** [review]
It looks like some of the files might need to be updated for 2022 copyright.  Other than that it looks good.

**Comment by HaoYang670** [comment]
Thank you. I will update them

**Comment by HaoYang670** [comment]
build

**Comment by HaoYang670** [comment]
build

**Review by revans2 — APPROVED** [review]

**Pull request merged**